build.sh

```diff
@@ -69,6 +69,7 @@ environment variables:
   XCMODE: xcodebuild (default), xcpretty or xctool
   CONFIGURATION: Debug or Release (default)
   REALM_CORE_VERSION: version in x.y.z format or "current" to use local build
+  REALM_EXTRA_BUILD_ARGUMENTS: additional arguments to pass to the build tool
```

Comment:
Yes. I was concerned that if it was always set for the `verify` targets then it'd be easier to accidentally end up with it being slipped into a releasability pipeline job, while setting things in the jenkins PR job can never effect the built products for the actual release.